source/docs/advanced/theme/wysiwyg.md

```diff
@@ -4,116 +4,385 @@ title: Display WYSIWYG content
-* `<widget>` tags are transformed `theme/modules/Wysiwyg/MagentoWysiwyg/Widget` components. However, you shouldn't write a `<widget>` tag manually. It is in fact coming from the `{% raw %}{{widget}}{% endraw %}` shortcode.
+**Add a custom Magento Widget**
 
-#### Add a custom Magento Widget
+Custom widgets will be automatically parsed. However, you will still need to map the widget's `type` to custom React Components. If you don't, they will be ignored and nothing will be rendered in the final page.
 
-Custom widgets will be automatically parsed. However, you will still need to map the widget's `type` to custom React Components. If you don't, they will be ignored.
+To do so, please override `theme/modules/WysiwygV2/MagentoWysiwyg/Widget/getWidgetComponent.js` in your own theme.
 
-To do so, please override `theme/modules/Wysiwyg/MagentoWysiwyg/Widget.js` in your own theme.
-
-```
-mkdir -p src/web/theme/modules/Wysiwyg/MagentoWysiwyg/
-cp node_modules/front-commerce/src/web/theme/modules/Wysiwyg/MagentoWysiwyg/Widget.js \
-  src/web/theme/modules/Wysiwyg/MagentoWysiwyg/Widget.js
-```
-
-Now, within your newly created `Widget.js` file, you will be able to add your own behavior. For instance, if you've created a widget in Magento with the type `acme/product-preview`, you will need to update the `Widget.js` file likewise:
+Now, within your newly created `getWidgetComponent.js` file, you will be able to add your own behavior. For instance, if you've created a widget in Magento that uses the type `acme/product-preview`, you will need to update the `Widget.js` file likewise:
 
 ```diff
+-const defaultWidgetsMap = {};
 +import React from "react";
-import PropTypes from "prop-types";
-+//import your matching React Component
-+import ProductPreview from "theme/modules/ProductPreview";
-
-const Widget = ({ type, attributes }) => {
-+  // And match the type with your component
-+  if (type === "acme/product-preview") {
-+    return <ProductPreview sku={attributes.sku} />
-+  }
++import Spinner from "theme/components/atoms/Spinner";
++import loadable from "@loadable/component";
 +
-  return null;
-};
++const defaultWidgetsMap = {
++  "acme/product-preview": loadable(() => import("./path/to/ProductPreview.js"), {
++    fallback: <Spinner />
++  })
++};
 
-Widget.propTypes = {
-  type: PropTypes.string.isRequired,
-  attributes: PropTypes.object.isRequired
-};
+// ... rest of the file is intact
+```
+
+If you restart your application, you will notice that the new widget component is displayed. Hurray!
+
+However, in most cases, you will need to fetch data to display all the needed information in your widget. For instance, if the widget is `{% raw %}{{ widget type="acme/product-preview" sku="VSK12" }}{% endraw %}` you will want to fetch the product associated with the given SKU. This is the same concept as for the `WysiwygNodeData` but one level deeper in the GraphQL schema.
```

Comment:
This section feels like explaining the previous part again.
We may add a rough schema or a bold sentence highlighting the difference, otherwise I feel that readers might overlook this section.

Reply:
I suggest that we look into it later.